### DISCUSSION TITLE
Should justified `Image`s stretch to fill up the available space?

### DISCUSSION THREAD

**Discussion by tangmi** [emilk/egui#101] [Q&A] [answered]
I noticed that an `egui::Image` inside a justified context will be stretched on the axis cross to the layout axis. This happens because when `Image` draws, it fills out the rectangle returned by `Layout::justify_or_align` (which itself is called inside `Ui::allocate_space_impl`).

For example, the following:

```rust
ui.with_layout(
    egui::Layout::top_down_justified(egui::Align::left()),
    |ui| {
        ui.image(self.texture_id, self.texture_actual_size);
    },
);
```

results in the image at the bottom of this post.

I'm writing this as a discussion and not an issue because I'm unsure if this is intended behavior or not (this is easy to work around, so I'm happy with a decision either way). As an anecdote, this was confusing to me when I switched from a horizontal layout to using `Ui::columns` (which justifies each of the resulting columns). I think I would personally expect justified images to just be placed on their alignment, rather than being stretched.

Anyways, thank you for working on such a cool project, I'm glad I discovered it!

![image](https://user-images.githubusercontent.com/2287247/103981182-2f627e80-5136-11eb-9ef9-6e43f014267c.png)

**Comment by emilk** [comment, marked as answer]
Hi!

I think it makes sense for images to keep their aspect ratios. I've pushed a fix in https://github.com/emilk/egui/commit/d344c9d9a3b08659d1edf01bb83624b3bf1029ab

**Reply by tangmi** [reply]
Wonderful, thanks for the quick response!
